modules/las/src/lib/laslaz-decoder.ts

```diff
@@ -78,37 +96,66 @@ function parseLASHeader(arraybuffer) {
   o.maxs = [bounds[0], bounds[2], bounds[4]];
   o.mins = [bounds[1], bounds[3], bounds[5]];
 
-  return o;
+  return o as LASHeader;
 }
 
 // LAS Loader
 // Loads uncompressed files
 //
 class LASLoader {
-  constructor(arraybuffer) {
+  arraybuffer: ArrayBuffer;
+  readOffset: number;
+  header: LASHeader;
+
+  constructor(arraybuffer: ArrayBuffer) {
     this.arraybuffer = arraybuffer;
+    this.readOffset = 0;
+    this.header = {
+      pointsOffset: 0,
+      pointsFormatId: 0,
+      pointsStructSize: 0,
+      pointsCount: 0,
+      scale: 0,
+      offset: 0,
+      maxs: [0],
+      mins: [0],
+      totalToRead: 0,
+      totalRead: 0,
+      versionAsString: '',
+      isCompressed: true
+    };
   }
-
+  /**
+   * Nothing needs to be done to open this
+   * @returns boolean
+   */
   open() {
-    // nothing needs to be done to open this file
-    //
     this.readOffset = 0;
     return true;
   }
-
+  /**
+   * Parsing of incoming binary
+   * @returns LASHeader
+   */
   getHeader() {
     this.header = parseLASHeader(this.arraybuffer);
     return this.header;
   }
 
-  readData(count, offset, skip) {
+  /**
+   * Reading data
+   * @param count
+   * @param skip
+   * @returns new ArrayBuffer, count, hasMoreData
+   */
+  readData(count: number, skip: number) {
     const {header, arraybuffer} = this;
-    if (!header) {
+    if (!this.header) {
```

Comment:
Why do you need `this.header` here? It is already taken from context above.

Reply:
> Why do you need `this.header` here? It is already taken from context above.

fixed